internal/services/automation/automation_runbook_resource.go

```diff
@@ -330,31 +341,15 @@ func resourceAutomationRunbookCreateUpdate(d *pluginsdk.ResourceData, meta inter
 
 	d.SetId(id.ID())
 
-	automationAccountId := jobschedule.NewAutomationAccountID(subscriptionID, id.ResourceGroupName, id.AutomationAccountName)
-
-	jsIterator, err := jsClient.ListByAutomationAccountComplete(ctx, automationAccountId, jobschedule.ListByAutomationAccountOperationOptions{})
-	if err != nil {
-		return fmt.Errorf("loading Automation Account %q Job Schedule List: %+v", id.AutomationAccountName, err)
-	}
-
-	for _, item := range jsIterator.Items {
-		if itemProps := item.Properties; itemProps != nil {
-			if itemProps.Runbook != nil && itemProps.Runbook.Name != nil && *itemProps.Runbook.Name == id.RunbookName {
-				if itemProps.JobScheduleId == nil || *itemProps.JobScheduleId == "" {
-					return fmt.Errorf("job schedule Id is nil or empty listed by %s Job Schedule List: %+v", id, err)
-				}
-				parsedId := jobschedule.NewJobScheduleID(subscriptionID, id.ResourceGroupName, id.AutomationAccountName, *itemProps.JobScheduleId)
-				if resp, err := jsClient.Delete(ctx, parsedId); err != nil {
-					if !response.WasNotFound(resp.HttpResponse) {
-						return fmt.Errorf("deleting job schedule Id listed by %s Job Schedule List:%v", id, err)
-					}
-				}
-			}
+	// **don't need** to list job schedules and delete all of them. update the runbook will recreate these job schedules automatically,
```

Comment:
Comparing to the original implementation, what's the main difference here? From what I see, the list query is different where you have specified an additional filter for only the linked `runbook` name.

Reply:
Can we modify the comment here a bit to state that once the `job_schedule` is changed, recreate them?